### ISSUE TITLE
Environment variable secrets fail Cloud Run deployment

### ISSUE THREAD

**Issue by kevmo314** [GoogleCloudPlatform/cloud-code-vscode#699]
Type: <b>Bug</b>

Create a secret in secret manager, link it to a Cloud Run instance from the cloud console.

When the Deploy on Cloud Run page loads in VS Code, it populates with invalid environment variables and the deploy fails.

- Managed Dependencies: on
- Cloud SDK Version: 412.0.0
- Skaffold Version: v2.0.3
- Minikube Version: 1.28.0

Extension version: 1.21.0
VS Code version: Code 1.70.2 (e4503b30fc78200f846c62cf8091b76ff5547662, 2022-08-16T05:35:13.448Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Connection to 'WSL: Ubuntu' could not be established  Canceled
Connection to 'WSL: Ubuntu' could not be established  Canceled
Connection to 'WSL: Ubuntu' could not be established  Canceled
Connection to 'SSH: muxer-2' could not be established  Canceled
Connection to 'WSL: Ubuntu' could not be established  Canceled
Remote OS version: Linux x64 5.15.79.1-microsoft-standard-WSL2
Remote OS version: Linux x64 5.15.79.1-microsoft-standard-WSL2
Remote OS version: Linux x64 5.15.79.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor             (24 x 3793)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.91GB (26.36GB free)|
|Process Argv|--crash-reporter-id 14d989d1-5fea-4fad-972d-68f8c301cd6f|
|Screen Reader|no|
|VM|0%|

Connection to 'WSL: Ubuntu' could not be established  Canceled

Connection to 'WSL: Ubuntu' could not be established  Canceled

Connection to 'WSL: Ubuntu' could not be established  Canceled

Connection to 'SSH: muxer-2' could not be established  Canceled

Connection to 'WSL: Ubuntu' could not be established  Canceled

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.79.1-microsoft-standard-WSL2|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor (24 x 3792)|
|Memory (System)|31.30GB (13.98GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.79.1-microsoft-standard-WSL2|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor (24 x 3792)|
|Memory (System)|31.30GB (13.98GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.79.1-microsoft-standard-WSL2|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor (24 x 3792)|
|Memory (System)|31.30GB (13.98GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
pyindex848cf:30577861
nodejswelcome1:30587005
282f8724:30602487
gswce1:30612156
iaj6b796:30613358
89544117:30613380
fim-prod:30623723

```

</details>

**Comment by davidponder** [comment]
Hello @kevmo314 !

Sorry to hear you're hitting this friction. I'll bounce this over to the team so we can get a fix out ASAP. Please let us know if 

Thanks!

**Comment by SKrupa** [comment]
This has been fixed as part of the 1.21.7 release, Cloud Code will now maintain the reference to your secret in the env vars. Please let us know if you face any further issues.
